Sources/Features/Shared/CloudKit/CloudKitOperation.swift

```diff
@@ -297,24 +328,54 @@ public class BatchedCloudKitOperation<T where T: NSOperation, T: CKBatchedOperat
     init<G where G: GeneratorType, G.Element == T>(timeout: NSTimeInterval? = 300, generator gen: G, enableBatchProcessing enable: Bool = true) {
 
         enableBatchProcessing = enable
-        generator = CloudKitOperationGenerator(timeout: timeout, generator: gen)
+
+        // Creates a CloudKitOperationGenerator object
+        let _generator = CloudKitOperationGenerator(timeout: timeout, generator: gen)
+
+        // Creates a Configuration Handler for next operations using the CloudKitOperationGenerator
+        let newConfigurationHandler: ConfigurationHandler = { [weak _generator] (nextOperation) -> PayLoad.ConfigureBlock? in
```

Comment:
I'd just capture this as `[unowned _generator]` and not worry about the necessary unwrapping:

```swift
guard let prepareForNextOperationHandler = _generator.prepareForNextOperationHandler else { return .None }
```